pkgs/test/cross/default.nix

```diff
@@ -91,6 +91,19 @@ let
       pkgFun = pkgs: pkgs.hello;
     };
 
+    pkg-config = {platformFun, crossPkgs, emulator}: pkgs.runCommand
+      "test-pkg-config-${crossPkgs.hostPlatform.config}"
+    {
+      depsBuildBuild = [ crossPkgs.pkgsBuildBuild.pkg-config ];
+      nativeBuildInputs = [ crossPkgs.pkgsBuildHost.pkg-config crossPkgs.pkgsBuildHost.zlib ];
+      depsBuildTarget = [ crossPkgs.pkgsBuildTarget.pkg-config ];
+      buildInputs = [ crossPkgs.zlib ];
+    } ''
+       mkdir $out
+       ${crossPkgs.pkgsBuildBuild.pkg-config.targetPrefix}pkg-config --cflags zlib > "$out/for-build"
+       ${crossPkgs.pkgsBuildHost.pkg-config.targetPrefix}pkg-config --cflags zlib > "$out/for-host"
+       !diff "$out/for-build" "$out/for-host"
```

Comment:
this looks like a mistake

Reply:
I wanted to see that they are different, though I should probably improve the test to raise things to a higher bar.

Reply:
what does ! do?

Reply:
negate the exit code, at least that is what I intend.

Reply:
Ah, I needed the space `! diff`